### PR TITLE
Fixes logging of target names with dashes

### PR DESCRIPTION
**Pull request by bkchr**
There was a bug in tracing-core which resulted in not supporting dashes
in target names. This was fixed upstream. Besides that a test was added
to ensure that we don't break this again.

Fixes: https://github.com/paritytech/substrate/issues/7251

### CODE REVIEW

File: client/cli/src/lib.rs [hunk 1]
```diff
@@ -371,4 +372,33 @@ mod tests {
 			assert!(test_filter("telemetry", Level::TRACE));
 		});
 	}
+
+	const EXPECTED_LOG_MESSAGE: &'static str = "yeah logging works as expected";
+
+	#[test]
+	fn dash_in_target_name_works() {
```
Comment (andresilva) [inline client/cli/src/lib.rs:379]: Nice test :)

File: client/cli/src/lib.rs [hunk 2]
```diff
@@ -371,4 +372,33 @@ mod tests {
 			assert!(test_filter("telemetry", Level::TRACE));
```
Comment (andresilva) [inline client/cli/src/lib.rs:372]: Maybe we can also add an example with dashes to this test 🤷

**Review by andresilva — APPROVED** [review]

File: client/cli/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply client/cli/src/lib.rs:372]: Added

**Comment by bkchr** [comment]
bot merge

**Review by tomaka — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

**Pull request merged**